src/neuroconv/utils/json_schema.py

```diff
@@ -60,6 +60,23 @@ def default(self, obj):
         return super().default(obj)
 
 
+class _NWBConversionOptionsEncoder(_NWBMetaDataEncoder):
+    """
+    Custom JSON encoder for conversion options of the data interfaces and converters (i.e. kwargs).
+
+    This encoder extends the default JSONEncoder class and provides custom serialization
+    for certain data types commonly used in interface source data.
+    """
+
+    def default(self, obj):
+
+        # Over-write behaviors for Paths
+        if isinstance(obj, Path):
+            return str(obj)
+
+        return super().default(obj)
```

Comment:
Why not add this to `_NWBMetaDataEncoder` instead of making a new class? I'm kind of surprised this isn't the default behavior for Path objs. What is the default behavior?

Reply:
The default behavior is to throw an error. 

Yeah, maybe we can use the same encoder for everything. Note that even before this PR we already had a `_NWBSourceDataEncoder`:

https://github.com/catalystneuro/neuroconv/blob/dcd248b3faec7f9b6f003784dbd899945b565df8/src/neuroconv/utils/json_schema.py#L46-L60

I think that the original idea was that metadata, source data and the conversion options might require different behavior so it was created that way. Check out the inheritance pattern here that kind of implements this simplification:

https://github.com/catalystneuro/neuroconv/pull/1142/files#r1856981408